### PR TITLE
feat(cli): improved run/preview display and --trace flag

### PR DESCRIPTION
**Pull request by lipikaramaswamy**
## Summary

Adds richer CLI output for the `run` and `preview` commands:

- `run` currently prints only `"Output written to: <path>"` — after this PR it prints a full summary block (entity counts by label, failed record details, rewrite evaluation metrics)
- `preview` currently dumps a raw `df.to_string()` — after this PR it prints readable per-record blocks with the key output fields
- New `--trace` flag on `run` writes the full `trace_dataframe` to a sidecar file

No new dependencies — all rendering uses stdlib + pandas.

## Plan

See [`plans/cli-display/plan.md`](plans/cli-display/plan.md) for the full implementation plan including output shapes, edge cases, function signatures, and test list.

## Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation update
- [ ] Refactoring

## Testing
- [ ] `make test` passes locally
- [ ] `make check` passes locally (format + lint + typecheck + lock-check)
- [ ] Added/updated tests for changes

## Documentation
- [ ] If docs changed: `make docs-build` passes locally

### CODE REVIEW

File: plans/cli-display/plan.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+    # Returns lines for one record block
+```
+
+Helpers return `list[str]` so they are unit-testable without capturing stdout.
+
+### `print_run_summary` output — replace mode
+
+```
+Output written to: /path/to/data_anonymized.csv
+Trace written to : /path/to/trace.csv          ← only when --trace given
```
Comment (lipikaramaswamy) [inline plans/cli-display/plan.md:73]: say "trace dataset"

File: plans/cli-display/plan.md [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+--- Summary ---
+Records processed : 1,234
+Failed records    : 0
+Total entities    : 4,521
+  first_name      :   891
+  email           :   312
+  ssn             :    47
+  ... and 3 more
+```
```
Comment (memadi-nv) [inline plans/cli-display/plan.md:83]: It's good to add some context about *what happened* besides *what the user ran*.  sth like:

```
Mode : replace (redact) 
Source : data.csv (1,234 rows) 
Output : /path/to/data_anonymized.csv (1,234 rows, 1.2 MB) 
Trace : /path/to/trace.csv ← only when --trace given Elapsed : 1m 23s
```

File: plans/cli-display/plan.md [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+Avg leakage mass     : 0.03
+Avg weighted leakage : 0.01
+Needs human review   : 12 (1.0%)
+```
+
+### Failed records block (appended when failures > 0)
+
+```
+--- Failed Records (3) ---
+  record_id=abc123  step=detect   reason=LLM timeout
```
Comment (memadi-nv) [inline plans/cli-display/plan.md:103]: Can we make this as the group failures? sth like Group by (step, reason, count):
```
--- Failed Records (12) ---
  detect    LLM timeout         × 8
  replace   JSON parse error    × 3
  detect    GLiNER OOM          × 1
```

Tip: run with -`-debug` to see per-record stack traces, or `--failed-output failed.csv` to export failures for triage.
Add `--failed-output PATH` (parallel to --trace): writes a small dataframe of record_id, step, reason for downstream triage.
The "Tip" footer turns a wall of text into a next action.

File: plans/cli-display/plan.md [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+### No `--trace` on `preview`
+
+Preview is a quick inspection tool. Trace export belongs to `run`.
+
+### Updated imports from `_output`
+
+```python
+from anonymizer.interface.cli._output import print_preview, print_run_summary, write_result, write_trace
+```
+
```
Comment (memadi-nv) [inline plans/cli-display/plan.md:196]: Adding some TTY-aware visual polish could be very helpful to distinguish different sections of the output in CLI. The plan's mockups are pure ASCII, which is fine — but a few cheap conditionals (no new deps) make the output much more scannable:
This could be sth like, green for the output path line, red for the failures count when > 0, dim for label rows. Bold section headers (--- Summary ---) so they don't disappear in long terminal scrollback.
Auto-disable on non-TTY, NO_COLOR env, or adding a --no-color flag.
